pkg/envoy/sds/response.go

```diff
@@ -34,14 +39,130 @@ func NewResponse(meshCatalog catalog.MeshCataloger, proxy *envoy.Proxy, request
 		log.Error().Err(err).Str("proxy", proxy.String()).Msgf("Error issuing a certificate for proxy")
 		return nil, err
 	}
-
 	// Set service certificate for this proxy
 	builder.SetCertificate(cert)
-	// Set proxy to add context to logging in builder methods
-	builder.SetProxy(proxy.String())
 
-	// TODO: decouple meshcatalog
-	return builder.Build(requestedCerts)
+	// Set trust domain
+	builder.SetTrustDomain(certManager.GetTrustDomain())
+
+	// getSDSSecrets (cert, requestedCerts, proxy)
+	// Create envoy secret resources based on the requested certs in the DiscoveryRequest
+	envoyCerts := getSDSSecrets(cert, requestedCerts, proxy, meshCatalog, builder.TrustDomain)
```

Comment:
In this format the responsibility of building the SDS config is mixed between `NewResponse`, and the builder.

We should maintain single responsbility, meaning that we should set the `requestedCerts`, and the `cert` object on the builder, then the builder object deals with the `getSDSSecrets` in the build method

Reply:
I have been attempting to convert `getRootCert` into a builder method and focused on decoupling `meshCatalog` but due to the difficulty of untangling it I realized that `getSDSSecrets` can remain as a composer. 

Calling `getSDSSecrets` from the Builder fits the builder pattern?

Reply:
I think the way it would work (based on the feedback I've received for RDS) is that getSDSSecrets and getRootCert would be methods in builder.go. The builder struct would have cert, requestedCerts, proxy, meshcatalog.ListServiceIdentitiesForService (we don't want to pass in the entire meshcatalog object, just what's necessary), and trustDomain as fields, and we set those when we create the builder object. 

Then, builder.Build() will call the getSDSSecrets and getRootCert methods and return the sdsResources.

Reply:
@nshankar13 that helped a lot! I did a spike attempting to get the results from `meshCatalog` before passing it to the builder but that turnt out messy.